src/main/java/Group15/MuscleSelectionView.java

```diff
@@ -0,0 +1,124 @@
+        equipmentSelector.getItems().addAll(
+                "All",
+                "Body weight",
+                "Barbell",
+                "Dumbbell",
+                "Machine"
+        );
+        equipmentSelector.setMinSize(200,50);
+        equipmentSelector.setPromptText("Select Equipment...");
+        String selectedEquipment = (String) equipmentSelector.getValue();
+
+        GridPane gridPane = new GridPane();
+        gridPane.setHgap(20);
+        gridPane.setVgap(20);
+        gridPane.setAlignment(Pos.CENTER);
+
+        List<ToggleButton> toggleButtons = new ArrayList<>();
+
+        int col = 0;
+        int row = 0;
+        for (BodyPart bodyPart : BodyPart.values()) {
+            Image bodyPartImage = new Image(Objects.requireNonNull(MuscleSelectionView.class.getResourceAsStream("/bodyparts/" +bodyPart.toString()+ ".png")));
+            ImageView bodyPartImageView = new ImageView(bodyPartImage);
+            bodyPartImageView.setFitHeight(60);
+            bodyPartImageView.setFitWidth((double) 200 /2);
+            bodyPartImageView.setPreserveRatio(true);
+
+            Label bodyPartName = new Label(bodyPart.toString());
+            bodyPartName.setMinWidth((double) 200 /3);
+            bodyPartName.setAlignment(Pos.CENTER);
+
+            imageAndTextButtons = new HBox(bodyPartImageView, bodyPartName);
+            imageAndTextButtons.setAlignment(Pos.CENTER);
+            imageAndTextButtons.setSpacing(10);
+
+
+            ToggleButton toggleButton = new ToggleButton();
+            toggleButton.setGraphic(imageAndTextButtons);
+            toggleButton.setMinSize(200,50);
+            toggleButtons.add(toggleButton);
+            gridPane.add(toggleButton, col, row);
+
+            col++;
+            if(col > 1) {
+                col = 0;
+                row++;
+            }
+        }
+
+        Button submitButton = new Button("Submit");
+
+        Button backButton = new Button("Back");
+        TextField inputField = new TextField();
+
+        submitButton.setPrefSize(200,50);
+        backButton.setPrefSize(200,50);
+
+        inputField.setPrefSize(200, 50);
+        inputField.setMaxSize(200, 50);
+        inputField.setPromptText("Input how many minutes to workout");
+
+        VBox.setMargin(submitButton, new Insets(50, 0, 0, 0));
+
+        inputAndEquipBox.getChildren().addAll(inputField,equipmentSelector);
+        inputAndEquipBox.setSpacing(20);
+        inputAndEquipBox.setAlignment(Pos.CENTER);
+
+        vBox.getChildren().add(gridPane);
+        vBox.getChildren().add(inputAndEquipBox);
+        vBox.getChildren().add(submitButton);
+        vBox.getChildren().add(backButton);
+
+        WorkoutAlgorithm workoutAlgorithm = new WorkoutAlgorithm();
+
+        submitButton.setOnAction(_ -> {
+            if(inputField.getText() == null || inputField.getText().isEmpty() || !inputField.getText().matches("\\d+")) {
+                System.out.println("Invalid input");
```

Comment:
Even though you are printing Invalid input, you still dont do anything about it, you should add a return